### PR TITLE
Port RPC cache to sqlite branch

### PR DESCRIPTION
**Pull request by tawmaz**
- Change RPCCache to unique_ptr type
 - Update list of cached RPC requests

### CODE REVIEW

File: src/rpc/cache.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class RPCCache
+{
+private:
+    Mutex CacheMutex;
+    std::map<std::string, UniValue> m_cache;
+    int m_blockHeight;
+    int m_cacheSize;
+    int m_maxCacheSize;
+    const std::vector<std::string> supportedMethods = {"getlastcomments",
```
Comment (andyoknen) [inline src/rpc/cache.h]: Please expand the list of cached methods:
```
getlastcomments
getcomments
getuseraddress
getaddressregistration
search
searchlinks
searchusers
gettags
getrawtransactionwithmessagebyid
getrawtransactionwithmessage
getrawtransaction
getusercontents
gethistoricalfeed
gethistoricalstrip
gethierarchicalfeed
gethierarchicalstrip
gethotposts
getuserprofile
getprofilefeed
getsubscribesfeed
txunspent
getaddressid
getuserstate
getpagescores
getcontent
getcontents
getaccountsetting
getcontentsstatistic
getusersubscribes
getusersubscribers
getuserblockings
getaddressscores
getpostscores
getpagescores
getstatisticbyhours
getstatisticbydays
getstatisticcontentbyhours
getstatisticcontentbydays
getrecomendedaccountsbytags
getrecomendedcontentsbyscoresonsimilarcontents
getaddressinfo
getcompactblock
getlastblocks
searchbyhash
gettransactions
getaddresstransactions
getblocktransactions
getbalancehistory
getcoininfo
estimatesmartfee
```

Comment (tawmaz) [reply src/rpc/cache.h]: Added a new commit to this PR with the updated list of RPC commands.

**Review by andyoknen — APPROVED** [review]

File: src/rpc/cache.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+#include <rpc/cache.h>
+#include <rpc/server.h>
+
+static const unsigned int MAX_CACHE_SIZE_MB = 64;
+
+RPCCache::RPCCache() 
+{
+    m_blockHeight = chainActive.Height();
+    m_maxCacheSize = gArgs.GetArg("-rpccachesize", MAX_CACHE_SIZE_MB) * 1024 * 1024;
```
Comment (only1question) [inline src/rpc/cache.cpp:13]: Not limited from above?

**Review by only1question — APPROVED** [review]

File: src/rpc/cache.cpp (hunk 2 above)
Comment (andyoknen) [reply src/rpc/cache.cpp:13]: This is a setup - in general, if a person wants to expose an indecent lot, then why not

**Pull request merged**